packages/snap/src/entities/snap.ts

```diff
@@ -182,4 +183,11 @@ export type SnapClient = {
    * @returns the user's preferences.
    */
   getPreferences(): Promise<GetPreferencesResult>;
+
+  /**
+   * Track events tha comply with the SIP-32 spec
+   *
+   * @param params parameters for the event to track
+   */
+  emitTrackingEvent(params: TrackEventParams): Promise<void>;
```

Comment:
Great! Do you think we can a bit stricter on the types where this function accepts an `event` (`enum`), the account (`BitcoinAccount`) and an optional `properties` (`Json`)?
This allows to enforce the event types, extract the needed data from the account and not rely on the externally defined `TrackEventParams` type (`Json` is fine).

I think we can forget about the `sensitiveProperties` for now unless you find something that is sensitive.